src/lib/storage/table.cpp

```diff
@@ -16,54 +16,66 @@
 #include "utils/assert.hpp"
 
 namespace opossum {
+Table::Table(uint32_t chunk_size) : _max_chunk_size(chunk_size) { _chunks.push_back(std::make_shared<Chunk>()); }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(row_count() == 0, "The table already contains data so adding a column is not possible.");
```

Comment:
Why should this not be possible? We can add the column for each existing chunk!
```
    for (auto const& type : _column_types)
      _chunks[_lastChunk]->add_segment(make_shared_by_data_type<BaseSegment, ValueSegment>(type));
```

Reply:
No this should not be possible as stated in the [`table.hpp`](https://github.com/youri-k/DYOD_WS1920/blob/master/src/lib/storage/table.hpp) and piazza. The reason is that we don't supply default values therefore the existing rows would have stuff missing.

```// adds a column to the end, i.e., right, of the table
  // this can only be done if the table does not yet have any entries, because we would otherwise have to deal
  // with default values
  void add_column(const std::string& name, const std::string& type);```

Reply:
Not having any entries and the amount of segments are two different things. Because we add empty segments anyway. The segments across **all** chunks form a column. That means that whenever we are adding a column, we need to add the corresponding segment that represents a chunk's share in that column to each chunk. That is where @Till-Later is right. 

@youri-k you are right about the fact that we do not allow adding a column after rows have been added, because in that case we would need to think about default values.

Reply:
> Not having any entries and the amount of segments are two different things.
That is correct but when we have 0 rows, we only have one chunk. Therefore **all** chunks mean the first chunk in this case.

Reply:
Oh you are right lol.